src/EventLoop.cpp

```diff
@@ -171,18 +182,72 @@ auto EventLoop::StartRequest(
     if (m_is_stopping) {
         return false;
     }
-
     // We'll prepare now since it won't block the event loop thread.
     request->prepareForPerform();
     {
-        std::lock_guard<std::mutex> guard { m_pending_requests_lock };
+        std::lock_guard<std::mutex> guard {m_pending_requests_mutex };
         m_pending_requests.emplace_back(std::move(request));
     }
     uv_async_send(&m_async);
-
     return true;
 }
 
+auto EventLoop::stopTimedOutRequests() -> void
+{
+    // If the wrappers multiset is empty, then we don't have anything to stop
+    if (m_response_wait_time_wrappers.empty())
+    {
+        return;
+    }
+
+    // Get the current timepoint from libuv (it caches the current timepoint at the beginning
+    // of every cycle through the event loop, so this should not be expensive).
+    auto now = uv_now(m_loop);
+
+    for (
+        auto current_wrapper = m_response_wait_time_wrappers.begin();
+        current_wrapper != m_response_wait_time_wrappers.end() && current_wrapper->GetData().m_timeout_time <= now;
+        // We increment current_wrapper by setting it to the iterator returned from erasing the current one
+    )
+    {
+        // Get the shared pointer out of the ResponseWaitTimeWrapper and into a unique pointer so it cleans
+        // itself up at the end of this method.
+        auto& data = current_wrapper->GetData();
+        auto shared_request = std::unique_ptr<std::shared_ptr<SharedRequest>>(data.m_shared_request_ptr_pointer);
```

Comment:
This is a bit confusing. Does the ResponseWaitTimeWrapper own the pointer?
If so, shouldn't it _not_ be a raw pointer?

Reply:
Yes, the wrapper should hold the shared pointer itself